R/common.R

```diff
@@ -988,3 +988,77 @@ findBestAlternateScenarios <- function(dfAlternateFeat,
   dfOptResult
 }
 
+#' @title
+#' Generate ROC curve for a dataset.
+#' @description Generates ROC curve and AUC for Sensitivity/Specificity or 
+#' Precision/Recall.
+#' @param predictions A vector of predictions from a machine learning model.
+#' @param labels A vector of the true labels. Must be the same length as 
+#' predictions.
+#' @param aucType Can be "SS" or "PR". Defaults to SS.
+#' @param plotFlg Binary value controlling plots Defaults to FALSE (no).
+#' @return auc Integral AUC of ROC plot.
+#'
+#' @import ROCR
+#' @export
+#' @references \url{http://healthcare.ai}
+#' @seealso \code{\link{healthcareai}}
+#' @examples
+#' 
+#' # generate data
+#' df <- data.frame(a = rep( seq(0,1,by=0.1), times=5),
+#'                  b = rep(0,times=55))
+#' # add positive labels
+#' df[c(seq(11,55,by=11)),2] <- 1
+#' df[c(seq(10,55,by=11)),2] <- 1
+#' df[c(seq(9,55,by=22)),2] <- 1
+#' df[c(4, 18, 30, 39, 52 ),2 ] <- 1
```

Comment:
Is this mostly random? Perhaps we could consolidate line 1012-1015 via something like 

df[sample(nrow(df), numberOfRowsToChange), 'b'] <- 1

Using col name ('b') would help make more explicit--need to pull all our code in that direction.